Bazel Basic test failing due to Bazel bug

https://grpc-testing.appspot.com/job/gRPC_pull_requests_Bazel_basic/2812/console

In particular.
```
ERROR: /var/local/git/grpc/src/proto/grpc/testing/BUILD:36:1: every rule of type proto_library implicitly depends upon the target '@com_google_protobuf//:protoc', but this target could not be found because of: no such package '@com_google_protobuf//': error loading package 'external': The repository named 'com_google_protobuf' could not be resolved.
```

@nicolasnoble believes it may be a manifestation of https://github.com/bazelbuild/bazel/issues/2682

Suggested workaround: pinning a previous working version of bazel in 
https://github.com/grpc/grpc/blob/master/tools/dockerfile/test/bazel/Dockerfile#L73

An interaction I don't quite understand:

If I try to pin Bazel to 0.4.4 via this
```
#========================
# Bazel installation
RUN apt-get install -y software-properties-common g++
RUN echo "deb [arch=amd64] http://storage.googleapis.com/bazel-apt stable jdk1.8" > /etc/apt/sources.list.d/bazel.list
RUN apt-get -y update
RUN apt-get -y install openjdk-8-jdk

# Pin Bazel to 0.4.4
RUN curl -fSsL -O https://github.com/bazelbuild/bazel/releases/download/0.4.4/bazel-0.4.4-installer-linux-x86_64.sh
RUN chmod +x ./bazel-0.4.4-installer-linux-x86_64.sh
RUN ./bazel-0.4.4-installer-linux-x86_64.sh
```
I get the following error when trying to run `tools/jenkins/run_bazel_basic.sh`:
```
____[18,349 / 19,429] Compiling src/core/ext/census/base_resources.c
____[18,351 / 19,429] Compiling src/core/ext/census/placeholders.c
____[18,416 / 19,429] Compiling src/core/ext/filters/client_channel/subchannel_index.c
____[18,428 / 19,429] Compiling src/core/lib/support/tmpfile_posix.c
ERROR: /var/local/git/grpc/BUILD:431:1: C++ compilation of rule '//:grpc_base' failed: gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -MD -MF ... (remaining 34 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
src/core/lib/compression/message_compress.c:41:18: fatal error: zlib.h: No such file or directory
compilation terminated.
____Building complete.
____Elapsed time: 79.257s, Critical Path: 39.61s
+ FAILED=true
+ '[' '' '!=' '' ']'
+ docker rm -f build_and_run_docker_d0cf6914-4d7d-4588-8a9e-fa693d931a05
build_and_run_docker_d0cf6914-4d7d-4588-8a9e-fa693d931a05
+ '[' true '!=' '' ']'
+ exit 1
```
but...
```
#========================
# Bazel installation
RUN apt-get install -y software-properties-common g++
RUN echo "deb [arch=amd64] http://storage.googleapis.com/bazel-apt stable jdk1.8" > /etc/apt/sources.list.d/bazel.list
RUN curl https://bazel.build/bazel-release.pub.gpg | apt-key add -
RUN apt-get -y update
RUN apt-get -y install bazel

# Pin Bazel to 0.4.4
# Installing Bazel via apt-get first is required before installing 0.4.4 to
# allow gRPC to build without errors. See https://github.com/grpc/grpc/issues/10553
RUN curl -fSsL -O https://github.com/bazelbuild/bazel/releases/download/0.4.4/bazel-0.4.4-installer-linux-x86_64.sh
RUN chmod +x ./bazel-0.4.4-installer-linux-x86_64.sh
RUN ./bazel-0.4.4-installer-linux-x86_64.sh
```
Trying to install Bazel to 0.4.4 after installing Bazel via apt-get allows me to run `tools/jenkins/run_bazel_basic.sh` without any issues.

--- a/tools/dockerfile/test/bazel/Dockerfile
+++ b/tools/dockerfile/test/bazel/Dockerfile
@@ -72,6 +72,13 @@ RUN curl https://bazel.build/bazel-release.pub.gpg | apt-key add -
 RUN apt-get -y update
 RUN apt-get -y install bazel
 
+# Pin Bazel to 0.4.4
+# Installing Bazel via apt-get first is required before installing 0.4.4 to
+# allow gRPC to build without errors. See https://github.com/grpc/grpc/issues/10553
+RUN curl -fSsL -O https://github.com/bazelbuild/bazel/releases/download/0.4.4/bazel-0.4.4-installer-linux-x86_64.sh
+RUN chmod +x ./bazel-0.4.4-installer-linux-x86_64.sh
+RUN ./bazel-0.4.4-installer-linux-x86_64.sh
+
 RUN mkdir -p /var/local/jenkins
 
 # Define the default command.

